cmd/customd/client/client.go

```diff
@@ -0,0 +1,448 @@
+package client
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/app"
+	"github.com/iov-one/weave/x/sigs"
+	"github.com/pkg/errors"
+	cmn "github.com/tendermint/tendermint/libs/common"
+	tmpubsub "github.com/tendermint/tendermint/libs/pubsub"
+	"github.com/tendermint/tendermint/rpc/client"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+type Header = tmtypes.Header
+type Status = ctypes.ResultStatus
+type GenesisDoc = tmtypes.GenesisDoc
+
+const BroadcastTxSyncDefaultTimeOut = 15 * time.Second
+
+var QueryNewBlockHeader = tmtypes.EventQueryNewBlockHeader
+
+// Client is an interface to interact with bcp
+type Client interface {
+	TendermintClient() client.Client
+	GetUser(addr weave.Address) (*UserResponse, error)
+	GetWallet(addr weave.Address) (*WalletResponse, error)
+	BroadcastTx(tx weave.Tx) BroadcastTxResponse
+	BroadcastTxAsync(tx weave.Tx, out chan<- BroadcastTxResponse)
+	BroadcastTxSync(tx weave.Tx, timeout time.Duration) BroadcastTxResponse
+	AbciQuery(path string, data []byte) (AbciResponse, error)
+}
+
+// CustomClient is a tendermint client wrapped to provide
+// simple access to the data structures used in bns.
+type CustomClient struct {
+	conn client.Client
+	// subscriber is a unique identifier for subscriptions
+	subscriber string
+}
+
+// NewClient wraps a CustomClient around an existing
+// tendermint client connection.
+func NewClient(conn client.Client) *CustomClient {
+	return &CustomClient{
+		conn: conn,
+		// TODO: make this random
```

Comment:
I am always against any TODO comments in the repository :)

Please either implement the missing functionality or leave a coming saying **why** it is implemented as it is and what is missing.

Reply:
According to git blame that code is written by @alpe. The question should be directed to him.

Reply:
@husio is right. Please remove the TODO comments. I did not contribute this code. It looks like what we had in the internal tools repo once.